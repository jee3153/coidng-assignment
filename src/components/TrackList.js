import React, { useContext } from 'react'
import DataContext from '../ContextAPI/DataContext'

import ImgPlaceHolder from '../images/no-image.jpg'

const TrackList = () => {

  const data = useContext(DataContext)

  const list = (data || []).map(track =>

    (
      <li className="track" key={track.id}>
        <div className="track-content__wrapper"  >
          <img className="track__artwork" src={track.artwork_url === null ? ImgPlaceHolder : track.artwork_url} alt="artwork" />
          <div className="track__rhs">
            <p className="track__title">{data[0].title}</p>
            <p className="track__username">{track.user.username}</p>
            <a className="track__link button__link button" href={track.user.permalink_url}>View more</a>
          </div>
        </div>
      </li>
    )
  )


  return (
    <div className="track__wrapper">
      <ul>{list}</ul>


    </div>
  )
}

export default TrackList
