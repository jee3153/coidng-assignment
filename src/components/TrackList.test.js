import React from 'react';
import renderer from 'react-test-renderer';

import TrackList from './TrackList';

describe('TrackList Snapshot', () => {
  test('renders', () => {
    const component = renderer.create(
      <TrackList />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
