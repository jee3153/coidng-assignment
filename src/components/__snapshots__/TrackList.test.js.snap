// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackList Snapshot renders 1`] = `
<div
  className="track__wrapper"
>
  <ul />
</div>
`;
